satellite/gracefulexit/getnodes_test.go

```diff
@@ -82,3 +82,107 @@ func TestGetExitingNodes(t *testing.T) {
 		}
 	})
 }
+
+func TestGetGracefulExitNodesByTimeframe(t *testing.T) {
+	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
+		ctx := testcontext.New(t)
+		defer ctx.Cleanup()
+
+		cache := db.OverlayCache()
+		exitingToday := make(map[storj.NodeID]bool)
+		exitingLastWeek := make(map[storj.NodeID]bool)
+		exitedToday := make(map[storj.NodeID]bool)
+		exitedLastWeek := make(map[storj.NodeID]bool)
+
+		now := time.Now()
+		lastWeek := time.Now().Add(-7 * 24 * time.Hour)
```

Comment:
This may also have weird failures around daylight savings time changes. I suggest `.AddDate()` instead of `.Add()`